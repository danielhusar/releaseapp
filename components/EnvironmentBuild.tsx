import React from 'react'
import { Heading, Box, Badge } from '@chakra-ui/core'
import Table from './Table'
import { IEnvironment } from '~/@types/environment'

const dateTimeFormat = new Intl.DateTimeFormat('en', {
  year: 'numeric',
  month: 'short',
  day: '2-digit',
  hour: 'numeric',
  minute: 'numeric',
  second: 'numeric',
  timeZone: 'America/Los_Angeles',
  timeZoneName: 'short',
})

export default function EnvironmentBuild({ aasm_state, last_deployed_at, current_build }: IEnvironment) {
  return (
    <Box
      p="6"
      borderWidth="1px"
      background="white"
      overflow="hidden"
      display="flex"
      flexDirection="column"
      data-testid="environment-info"
    >
      <Heading size="md" fontWeight={400} marginBottom={5}>
        Build
      </Heading>

      <Table slim={true}>
        <tr>
          <td>
            <b>Status</b> -{' '}
            <Badge variant="solid" variantColor={aasm_state === 'done' ? 'green' : 'red'}>
              {aasm_state}
            </Badge>
          </td>
        </tr>
        <tr>
          <td>
            <b>Created at</b> - {dateTimeFormat.format(current_build.created_at)}
          </td>
        </tr>
        <tr>
          <td>
            <b>Deployed at</b> - {dateTimeFormat.format(last_deployed_at)}
          </td>
        </tr>
        <tr>
          <td>
            <b>Branch</b> - {current_build.branch}
          </td>
        </tr>
        <tr>
          <td>
            <b>Commiter</b> - {current_build.commit_email}
          </td>
        </tr>
        <tr>
          <td>
            <b>Commit message</b> - {current_build.commit_message}
          </td>
        </tr>
      </Table>
    </Box>
  )
}
