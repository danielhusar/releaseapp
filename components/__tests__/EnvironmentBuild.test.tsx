import React from 'react'
import { render, environment } from './helpers'
import EnvironmentBuild from '../EnvironmentBuild'

describe('EnvironmentBuild', () => {
  it('renders', () => {
    const { container } = render(<EnvironmentBuild {...environment} />)
    expect(container).toMatchSnapshot()
  })
})
