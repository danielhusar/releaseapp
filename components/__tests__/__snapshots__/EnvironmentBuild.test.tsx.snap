// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvironmentBuild renders 1`] = `
.emotion-3 {
  padding: 1.5rem;
  border-width: 1px;
  background: white;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  font-size: 1.25rem;
  line-height: 1.25;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  margin-bottom: 1.25rem;
}

.emotion-2 {
  border: none;
  text-align: left;
  font-size: 0.875rem;
  width: 100%;
}

.emotion-2 th {
  padding: 20px 5px;
  font-weight: 500;
}

.emotion-2 td {
  border-top: 1px solid #ccc;
  padding: 30px 5px;
}

.emotion-2 td {
  padding: 15px 5px;
}

.emotion-1 {
  display: inline-block;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  text-transform: uppercase;
  font-size: 0.75rem;
  border-radius: 0.125rem;
  font-weight: 700;
  white-space: nowrap;
  vertical-align: middle;
  background-color: #38a169;
  color: #fff;
}

<div>
  <div
    class="emotion-3"
    data-testid="environment-info"
  >
    <h2
      class="emotion-0"
    >
      Build
    </h2>
    <table
      class="emotion-2"
    >
      <tbody>
        <tr>
          <td>
            <b>
              ID
            </b>
             - 
            10758
          </td>
        </tr>
        <tr>
          <td>
            <b>
              Status
            </b>
             -
             
            <div
              class="emotion-1"
            >
              done
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <b>
              Build at
            </b>
             - 
            Apr 08, 2020, 1:51:02 PM PDT
          </td>
        </tr>
        <tr>
          <td>
            <b>
              Branch
            </b>
             - 
            master
          </td>
        </tr>
        <tr>
          <td>
            <b>
              Commiter
            </b>
             - 
            tmcclung@gmail.com
          </td>
        </tr>
        <tr>
          <td>
            <b>
              Commit message
            </b>
             - 
            added run-configure.sh
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
